fe1-web/src/features/evoting/components/EventElection.tsx

```diff
@@ -86,77 +100,81 @@ const EventElection = (props: IPropTypes) => {
       });
   };
 
-  // Here we use the election object form the redux store in order to see the electionStatus
-  // update when an  incoming electionEnd or electionResult message comes
-  // (in handler/ElectionHandler.ts)
-  const getElectionDisplay = (status: ElectionStatus) => {
-    switch (status) {
-      case ElectionStatus.NOT_STARTED:
-        return (
-          <>
-            <SectionList
-              sections={questions}
-              keyExtractor={(item, index) => item + index}
-              renderSectionHeader={({ section: { title } }) => (
-                <Text style={styles.textQuestions}>{title}</Text>
-              )}
-              renderItem={({ item }) => <Text style={styles.textOptions}>{`\u2022 ${item}`}</Text>}
-            />
-            {isOrganizer && <WideButtonView title="Open election" onPress={onOpenElection} />}
-          </>
-        );
-      case ElectionStatus.OPENED:
-        return (
-          <>
-            {questions.map((q, idx) => (
-              <CheckboxList
-                key={q.title + idx.toString()}
-                title={q.title}
-                values={q.data}
-                onChange={(values: number[]) =>
-                  setSelectedBallots({ ...selectedBallots, [idx]: new Set(values) })
-                }
-              />
-            ))}
-            <WideButtonView title={STRINGS.cast_vote} onPress={onCastVote} />
-            <Badge value={hasVoted} status="success" />
-            {isOrganizer && (
-              <WideButtonView
-                title="Terminate Election / Tally Votes"
-                onPress={onTerminateElection}
-              />
-            )}
-          </>
-        );
-      case ElectionStatus.TERMINATED:
-        return (
-          <>
-            <Text style={styles.text}>Election Terminated</Text>
-            <Text style={styles.text}>Waiting for result</Text>
-          </>
-        );
-      case ElectionStatus.RESULT:
-        return (
-          <>
-            <Text style={styles.text}>Election Result</Text>
-            {election.questionResult &&
-              election.questionResult.map((question: QuestionResult) => (
-                <BarChartDisplay data={question.result} key={question.id.valueOf()} />
-              ))}
-          </>
-        );
-      default:
-        console.warn('Election Status was undefined in Election display', election);
-        return null;
-    }
-  };
+  const canCastVote = election.version !== ElectionVersion.SECRET_BALLOT || electionKey;
 
-  return (
-    <>
-      <TimeDisplay start={election.start.valueOf()} end={election.end.valueOf()} />
-      {getElectionDisplay(election.electionStatus)}
-    </>
-  );
+  switch (election.electionStatus) {
```

Comment:
this switch is getting kinda big. Should we split it in multiple sub-components ?

Reply:
I agree. Subcomponents are an option but will introduce a lot of redundant code since many parameters need to be passed to all of them but I it is also the best option I can think of at the moment